qualtran/cirq_interop/_cirq_to_bloq.py

```diff
@@ -204,39 +193,37 @@ def _wire_symbol_from_gate(
     return _cirq_wire_symbol_to_qualtran_wire_symbol(symbol, wire_reg.side)
 
 
-def _add_my_tensors_from_gate(
+def _my_tensors_from_gate(
     gate: cirq.Gate,
     signature: Signature,
-    short_name: str,
-    tn: 'qtn.TensorNetwork',
-    tag: Any,
     *,
-    incoming: Dict[str, 'SoquetT'],
-    outgoing: Dict[str, 'SoquetT'],
-):
+    incoming: Dict[str, 'ConnectionT'],
+    outgoing: Dict[str, 'ConnectionT'],
+) -> List['qtn.Tensor']:
     import quimb.tensor as qtn
 
-    from qualtran.simulation.tensor._tensor_data_manipulation import (
-        tensor_data_from_unitary_and_signature,
-    )
+    from qualtran.simulation.tensor._dense import _order_incoming_outgoing_indices
 
     if not cirq.has_unitary(gate):
         raise NotImplementedError(
-            f"CirqGateAsBloq.add_my_tensors is currently supported only for unitary gates. "
-            f"Found {gate}."
+            f"CirqGateAsBloq.my_tensors is only supported for unitary gates, not {gate}."
         )
-    unitary = tensor_data_from_unitary_and_signature(cirq.unitary(gate), signature)
-    incoming_list = [
-        *itertools.chain.from_iterable(
-            [np.array(incoming[reg.name]).flatten() for reg in signature.lefts()]
+
+    if any(reg.side != Side.THRU for reg in signature):
+        raise ValueError(
+            f"CirqGateAsBloq.my_tensors is only supported for "
+            f"gates with thru registers. Found {gate}."
         )
```

Comment:
This restriction was not present in the original `_add_my_tensors_from_gate` method; so it'll be nice to not add this restriction as part of this large scale refactoring. The `CirqGateAsBloq` specific errors should be raised in the implementation of `CirqGateAsBloq`; not in the implementation of ` _my_tensors_from_gate`. 

The original implementation works by first assuming that all registers are THRU registers and then extracting tensor data from the correct subspace whenever a register is not a THRU register. This is done by `tensor_data_from_unitary_and_signature` -- the method is still implemented but not used anymore here. 

Can you please revert the restricting changes to this method make sure the more general behavior is maintained? This method is called not just from `CirqGateAsBloq` but also from `GateWithRegisters`, that may have directional registers.

Reply:
this should really have a test if it's behavior we're relying on. I'll revert the changes, though